### PR TITLE
#5 Edit Booking init enhancement

### LINKED ISSUE #5: Feature - Edit Booking
* Frontend - Same process as Add - difference fetch request to get exisiting data. Fill form fields with data and update object on submit.
- Update date and time on Add feature.
- Add delete button to form -> Delete Modal

### PR DESCRIPTION
**Pull request by jpaea23**
Closes #5 , Resolves #5

### CODE REVIEW

File: frontend/src/components/SchedCalendar/CalendarContent/CalendarSched/CalendarSched.js [hunk 1]
```diff
@@ -35,14 +38,25 @@ const CalendarSched = (props) => {
...
+    deleteDetails = [
+      {'Id': jobId},
+      {'Name': clientName},
+      {'Number': clientNumber},
+      {'Address': jobAddress},
+    ];
+    editDetails = {
+      'jobId': jobId,
+      'clientId': client['clientId'],
+      'time': key + '',
```
Comment (damoster) [inline frontend/src/components/SchedCalendar/CalendarContent/CalendarSched/CalendarSched.js:56]: What is the reasoning behind concat with '' here?

File: frontend/src/containers/Calendar/Calendar.test.js [hunk 2]
```diff
@@ -1,18 +1,82 @@
...
+    "2020-05-01": {},
+    "2020-05-02": {}
+  }
+
+
+test('GenerateTimeslotArr', () => {
+  let arr = calcAvailTimeSlot(jobObjFound,selectDateFill);
+
+  const expected = [1, 2, 3, 4, 5, 6];
+  expect([4, 1, 6, 7, 3, 5, 7, 5, 4, 6]).not.toEqual(
```
Comment (damoster) [inline frontend/src/containers/Calendar/Calendar.test.js:40]: What's the point of this expect check?

File: frontend/src/containers/Calendar/Calendar.test.js [hunk 3]
```diff
@@ -1,18 +1,82 @@
...
+    expect.arrayContaining(expected),
+  );
+
+  //return type is array 
+  expect(arr).toEqual(expect.any(Array))
+  //return obj in array
+  expect(arr).toEqual(
+    expect.arrayContaining([
+      expect.objectContaining({
```
Comment (damoster) [inline frontend/src/containers/Calendar/Calendar.test.js:49]: Should do checks for expect.toMatchObject and compare to exact expected output. A containing check is not enough as what if some new change adds more than what it's expecting? This test would not fail and you now have unexpected behaviour

File: frontend/src/components/SchedCalendar/CalendarContent/CalendarSched/CalendarSched.js (hunk 1 above)
Comment (jpaea23) [reply frontend/src/components/SchedCalendar/CalendarContent/CalendarSched/CalendarSched.js:56]: converting to a string value without using the toString() method

File: frontend/src/containers/Calendar/Calendar.test.js (hunk 2 above)
Comment (jpaea23) [reply frontend/src/containers/Calendar/Calendar.test.js:40]: Oh i forgot to remove this test case. I was testing how this test case works lol

File: frontend/src/containers/Calendar/Calendar.test.js [hunk 4]
```diff
@@ -1,18 +1,67 @@
...
+  // return match array
+  expect(arr).toEqual([
+    {'7': {'jobId': 4, 'clientId': 1}},
+    {'9': {'jobId': 4, 'clientId': 1}},
+    11, 12, 13,
+    {'14': {'jobId': 7, 'clientId': 1}},
+    {'16': {'jobId': 7, 'clientId': 1}},
+  ]);
+
+  expect(arr).toEqual(expect.arrayContaining([11, 12, 13]));
```
Comment (damoster) [inline frontend/src/containers/Calendar/Calendar.test.js:55]: /nit don't need this line

File: frontend/src/containers/Calendar/Calendar.test.js [hunk 5]
```diff
@@ -1,18 +1,67 @@
...
+  expect(arr).toEqual(expect.not.arrayContaining([7, 8, 14, 15, 16, 15]));
+  expect(arr).toEqual(expect.not.arrayContaining([6, 18]));
+});
+
+test('EmptyTimeslotArr', () => {
+  const arr = calcAvailTimeSlot(jobObjEmpty, selectDateEmpty);
+
+  // check timeslots
+  // standard expected day timeslot
```
Comment (damoster) [inline frontend/src/containers/Calendar/Calendar.test.js:64]: /supernit could get rid of these comments, or combine them into one line

**Pull request merged**